features/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/TARGET_TFM/TARGET_TFM_V1_0/TARGET_TFM_DUALCPU/src/platform_ns_mailbox.c

```diff
@@ -120,145 +100,47 @@ int32_t tfm_ns_mailbox_hal_init(struct ns_mailbox_queue_t *queue)
-                 * generate a deadlock.
-                 * This delay won't harm performance too much since non-secure
-                 * core has to busy wait here anyway.
-                 * Alternatively, non-secure core can wait for release
-                 * notification event from secure core. However, it is more
-                 * complex and requires more code and more modifications.
-                 */
-                volatile uint32_t count = 1000;
-                while(count > 0) {
-                    count--;
-                }
-                Cy_IPC_Sema_Status(sema_num);
-            }
-        }
-
-        ret = CY_IPC_SEMA_SUCCESS;
-    }
-
-    return ret;
-}
-
-static cy_en_ipcsema_status_t mailbox_raw_spin_unlock(uint32_t ipc_channel,
-                                                      uint32_t sema_num)
-{
-    uint32_t semaIndex;
-    uint32_t semaMask;
-    cy_stc_ipc_sema_t *semaStruct;
-    cy_en_ipcdrv_status_t acqStatus;
-    cy_en_ipcsema_status_t ret = CY_IPC_SEMA_BAD_PARAM;
-    bool is_unlock = false;
-    IPC_STRUCT_Type *semaIpcStruct;
-
-    /* Get IPC register structure */
-    semaIpcStruct = Cy_IPC_Drv_GetIpcBaseAddress(ipc_channel);
-    /* Get pointer to structure */
-    semaStruct = (cy_stc_ipc_sema_t *)Cy_IPC_Drv_ReadDataValue(semaIpcStruct);
-
-    if (sema_num < semaStruct->maxSema) {
-        semaIndex = sema_num / CY_IPC_SEMA_PER_WORD;
-        semaMask = (uint32_t)(1ul << (sema_num - \
-                                      (semaIndex * CY_IPC_SEMA_PER_WORD)));
-
-        while (!is_unlock) {
-            /* Check to make sure the IPC channel is released
-               If so, check if specific channel can be locked. */
-            do {
-                acqStatus = Cy_IPC_Drv_LockAcquire(semaIpcStruct);
-            } while (acqStatus != CY_IPC_DRV_SUCCESS);
-
-            if ((semaStruct->arrayPtr[semaIndex] & semaMask) != 0ul) {
-                semaStruct->arrayPtr[semaIndex] &= ~semaMask;
-                is_unlock = true;
-            }
-
-            /* Release, but do not trigger a release event */
-            (void)Cy_IPC_Drv_LockRelease(semaIpcStruct,
-                                         CY_IPC_NO_NOTIFICATION);
-        }
-
-        ret = CY_IPC_SEMA_SUCCESS;
-    }
-
-    return ret;
-}
-
 void tfm_ns_mailbox_hal_enter_critical(void)
 {
     saved_irq_state = Cy_SysLib_EnterCriticalSection();
 
-    mailbox_raw_spin_lock(CY_IPC_CHAN_SEMA, MAILBOX_SEMAPHORE_NUM);
+    IPC_STRUCT_Type* ipc_struct =
+        Cy_IPC_Drv_GetIpcBaseAddress(IPC_PSA_MAILBOX_LOCK_CHAN);
+    while(CY_IPC_DRV_SUCCESS != Cy_IPC_Drv_LockAcquire (ipc_struct))
+    {
+    }
 }
 
 void tfm_ns_mailbox_hal_exit_critical(void)
 {
-    mailbox_raw_spin_unlock(CY_IPC_CHAN_SEMA, MAILBOX_SEMAPHORE_NUM);
-
+    IPC_STRUCT_Type* ipc_struct =
+        Cy_IPC_Drv_GetIpcBaseAddress(IPC_PSA_MAILBOX_LOCK_CHAN);
+        Cy_IPC_Drv_LockRelease(ipc_struct, CY_IPC_NO_NOTIFICATION);
```

Comment:
unexpected indent, similar in tfm_ns_mailbox_hal_exit_critical_isr

Reply:
fixed